### PR TITLE
Feature/remove original static names

### PR DESCRIPTION
**Pull request by lenarother**
Add storage modul to barbeque. It overrides url method to make it easier to work with comapss (implemented in other projects), and post_process (new) to  delete originally named static files to facilitate docker usage (smaller container size).

### CODE REVIEW

File: barbeque/static.py [hunk 1]
```diff
@@ -16,17 +28,58 @@ def __call__(self, request):
...
         return self.process_response(request, response)
 
+    def serve_response(self, request, file_path):
+        return serve(request, file_path, document_root=settings.STATIC_ROOT)
+
+    def find_requested_file_hashed_name(self, requested_path):
+        # Load static files mapping manifest.
+        # It maps original names with hashed ones.
+        storage = ManifestStaticFilesStorage()
+        manifest = storage.load_manifest()
```
Comment (andreashug) [inline barbeque/static.py]: The manifest is loaded and parsed every time this function is called. To improve performance I'd do this only once in the `__init__` function.

File: barbeque/static.py [hunk 2]
```diff
@@ -16,17 +28,58 @@ def __call__(self, request):
...
+        if manifest is None or len(manifest) == 0:
+            return None
+
+        # requested file has an original (not hashed) name
+        requested_path = requested_path.strip('/')
+        if requested_path in manifest:
+            return manifest[requested_path]
+
+        # requested file has an old hash in name
+        requested_path_parts = requested_path.split('.')
```
Comment (andreashug) [inline barbeque/static.py]: I think this would be more compact and explicit with a regex substitution like `re.sub(r'(\.[0-9a-f]{12}\.)?(\w+)$', r'.\2', requested_path)` (not tested yet) This way it could be applied to the path before in any way because if the hash is missing (i.e. it's the original file) it's a no-op. Having this in an extra function makes it easy to write some more tests for corner cases like files without extension or files with dots in the name.

File: barbeque/storage.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+from django.conf import settings
+from django.contrib.staticfiles.utils import matches_patterns
+from django.core.files.base import ContentFile
+from django.utils.encoding import force_bytes, force_text
+from django.utils.six import iteritems
+
+
+class StaticFilesStorage(ManifestStaticFilesStorage):
+    def url(self, name, force=False):
```
Comment (andreashug) [inline barbeque/storage.py]: This hack is project specific and shouldn't go into barbeque

File: barbeque/storage.py [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+from collections import OrderedDict
+
+from django.conf import settings
+from django.contrib.staticfiles.utils import matches_patterns
+from django.core.files.base import ContentFile
+from django.utils.encoding import force_bytes, force_text
+from django.utils.six import iteritems
+
+
+class StaticFilesStorage(ManifestStaticFilesStorage):
```
Comment (andreashug) [inline barbeque/storage.py]: I'd give this a more explicit name. What about CompactManifestStaticFileStorage?

File: barbeque/static.py [hunk 5]
```diff
@@ -16,17 +28,58 @@ def __call__(self, request):
         response = self.get_response(request)
         return self.process_response(request, response)
 
+    def serve_response(self, request, file_path):
+        return serve(request, file_path, document_root=settings.STATIC_ROOT)
+
+    def find_requested_file_hashed_name(self, requested_path):
+        # Load static files mapping manifest.
+        # It maps original names with hashed ones.
+        storage = ManifestStaticFilesStorage()
```
Comment (andreashug) [inline barbeque/static.py]: I'd fetch the storage from settings in case a custom one is used that handles the manifest file in another way.

File: barbeque/storage.py [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+        of our static storage. One could try to make it more generic by using
+        STATIC_URL but as we hardcoded "static" everywhere, we do it here too.
+        """
+        name_parts = name.split(os.sep)
+        if name_parts[0] == 'static':
+            name = os.sep.join(name_parts[1:])
+
+        return super(StaticFilesStorage, self).url(name, force=force)
+
+    def post_process(self, paths, dry_run=False, **options):
```
Comment (andreashug) [inline barbeque/storage.py]: I think it's risky to copy and modify this big function because it may change (already between 1.9 and master). Could the original files be deleted by using the manifest somehow? Let's talk about this in person =)

File: barbeque/static.py (hunk 5 above)
Comment (lenarother) [inline barbeque/static.py]: Done

File: barbeque/static.py (hunk 1 above)
Comment (lenarother) [inline barbeque/static.py]: Dane

File: barbeque/static.py (hunk 2 above)
Comment (lenarother) [inline barbeque/static.py]: Done

File: barbeque/storage.py (hunk 4 above)
Comment (lenarother) [inline barbeque/storage.py]: Done

File: barbeque/storage.py (hunk 3 above)
Comment (lenarother) [inline barbeque/storage.py]: Done

File: barbeque/storage.py (hunk 6 above)
Comment (lenarother) [inline barbeque/storage.py]: Done (deleting doesn't need try-except as it is not throwing error when file is not there: https://docs.djangoproject.com/el/1.10/ref/files/storage/)
